Cabal/Cabal.cabal

```diff
@@ -315,6 +315,7 @@ library
     Distribution.Compat.Exception
     Distribution.Compat.Graph
     Distribution.Compat.Internal.TempFile
+    Distribution.Compat.Prelude
```

Comment:
@ezyang we only need to import that `.Internal` module once, in the `Distribution.Client.Compat.Prelude` module, so we can tolerate a single `{-# OPTIONS_GHC -Wno-deprecations #-}` in there, can't we?